components/board/client.go

```diff
@@ -217,7 +218,7 @@ type analogReaderClient struct {
 }
 
 func (arc *analogReaderClient) Read(ctx context.Context, extra map[string]interface{}) (int, error) {
-	ext, err := protoutils.StructToStructPb(extra)
+	ext, err := data.GetExtraFromContext(ctx, extra)
```

Comment:
Spoke with Katherine offline about how we can enforce the FromDataManagement field gets passed into extra in the request for every single component, even as new components and component client implementations are added. One idea would be to add a client-side interceptor that looks for FromDataManagement in the context and injects that into extra.

Reply:
I added a UnaryClientInterceptor to the most recent commit - a major benefit is that this makes the client side of things more or less automatic (which means that the client code changes in this PR can be removed - yay!) but would love some feedback on this approach before I remove those changes.

Reply:
Update: removed client code changes - if anyone wants to see this approach instead of the interceptor one please look at the diff from the first commit!